pkg/operation/hybridbotanist/controlplane.go

```diff
@@ -446,11 +448,15 @@ func (b *HybridBotanist) DeployCloudControllerManager() error {
 			"checksum/configmap-cloud-provider-config":        b.CheckSums[common.CloudProviderConfigName],
 		},
 	}
-	cloudSpecificValues, err := b.ShootCloudBotanist.GenerateCloudControllerManagerConfig()
+	cloudSpecificValues, chartName, err := b.ShootCloudBotanist.GenerateCloudControllerManagerConfig()
 	if err != nil {
 		return err
 	}
 
+	if chartName == "" {
```

Comment:
Why this implicit logic? if you extend the interface to return a chartName then let AWS, GCP, etc. return the correct chart name instead of having this `if` condition here.